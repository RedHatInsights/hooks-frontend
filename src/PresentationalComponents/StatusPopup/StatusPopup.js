import React, { Fragment } from 'react';
import { Tooltip } from '@patternfly/react-core';
import PropTypes from 'prop-types';

export const StatusPopup = ({ children, status, lastAttempt, lastFailure }) => {
    let lines = [];
    if (status === 'success') {
        lines.push(`Last delivery attempt: ${ lastAttempt }`);
    } else if (status === 'failure') {
        lines.push(`Last delivery attempt: ${ lastAttempt }`);
        // Warning: Each child in an array or iterator should have a unique "key" prop.
        // What?
        lines.push(<br key={ '' } />);
        lines.push(`Failed at: ${ lastFailure }`);
    } else {
        lines.push('No delivery attemps so far');
    }

    return (
        <Fragment>
            <Tooltip
                content={ lines } >
                <span>
                    { children }
                </span>
            </Tooltip>
        </Fragment>
    );
};

StatusPopup.propTypes = {
    status: PropTypes.oneOf([ 'success', 'failure', 'unknown' ]).isRequired,
    lastAttempt: PropTypes.string,
    lastFailure: PropTypes.string,
    children: PropTypes.node
};

export default StatusPopup;
