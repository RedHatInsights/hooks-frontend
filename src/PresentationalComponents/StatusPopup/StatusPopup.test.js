import { StatusPopup } from './StatusPopup';

describe('StatusPopup', () => {
    it('expect to render when status is unknown', () => {
        const wrapper = shallow(
            <StatusPopup status={ 'unknown' } />
        );
        expect(wrapper.render()).toMatchSnapshot();
    });
});

describe('StatusPopup', () => {
    it('expect to render TimesCirclePopup', () => {
        const wrapper = shallow(
            <StatusPopup status={ 'failure' } />
        );
        expect(wrapper.render()).toMatchSnapshot();
    });
});

describe('StatusPopup', () => {
    it('expect to render QuestionCirclePopup', () => {
        const wrapper = shallow(
            <StatusPopup status={ 'success' } />
        );
        expect(wrapper.render()).toMatchSnapshot();
    });
});
