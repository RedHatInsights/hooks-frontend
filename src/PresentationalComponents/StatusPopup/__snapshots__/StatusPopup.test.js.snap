// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusPopup expect to render QuestionCirclePopup 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`StatusPopup expect to render TimesCirclePopup 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`StatusPopup expect to render when status is unknown 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
